packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-sign-in-pwd-email/ios/integrationsteps.md

```diff
@@ -1,5 +1,371 @@
+
+<div class="common-image-format">
+
+ ![Diagram showing the integration flow of the sample app and Swift SDK](/img/oie-embedded-sdk/oie-embedded-sdk-swift-sample-code-overview.png
+ "Swift sample and sdk diagram")
+
+</div>
+
+### Steps
+
+#### Step 1: Initialize MultifactorLogin
+
+The first step is to initialize `MultifactorLogin` and pass in a
+`configuration` object and `stephandler` function or code block. This
+`stephandler` allows you to interact with the authentication flow and
+gives you the ability to prompt the user to choose the email authenticator
+and subsequent verification code.
+
+ ```swift
+self.authHandler = MultifactorLogin(configuration: configuration)
+{ step in
+    switch step {
+    case .chooseFactor(let factors):
+        // Use this to prompt the user for the factor you'd like to authenticate with.
+        if factors.contains(.email) {
+            self.authHandler?.select(factor: .email)
+        }
+    case .verifyCode(factor: let factor):
+        // Prompt the user for the verification code; when they supply it, call the `verify` function.
+        if factor == .email {
+            self.authHandler?.verify(code: "123456")
+        }
+    }
+    case .chooseMethod(let methods):
+        // Use this to prompt the user for the method you'd like to authenticate with.
+        if methods.contains(.sms) {
+            self.authHandler?.select(factor: .phone,
+                                     method: .sms,
+                                     phoneNumber: "+15551234567")
+        }
+     }
+}
+```
+
+For more information on how to set the `configuration` parameter, see
+[SDK](/docs/guides/oie-embedded-common-download-setup-app/ios/main/#sdk)
+in
+[Download and set up the SDK, Sign-In Widget, and sample app](/docs/guides/oie-embedded-common-download-setup-app/ios/main)
+
+
+#### Step 2: Call the login method
+
+The next step is to call the `login` method passing in the username
+and password. The method can take a completion handler which
+will be invoked once, either with a fatal error, or success
+with a token.
+
+```swift
+ self.authHandler.login(username: "user@example.com",
+                       password: "secretPassword")
+{ result in
+    switch result {
+    case .success(let token):
+        print(token)
+    case .failure(let error):
+        print(error)
+    }
+}
+```
+
+#### Step 3: Get user profile information (optional)
+
+Optionally, you can obtain basic user information after a successful sign-in by
+making a request to Okta's Open ID Connect authorization server. See
+[Get user profile information after sign in](/docs/guides/oie-embedded-sdk-alternate-flows/ios/main/#get-user-profile-information-after-sign-in).
+
 ## Sample code
 
-Swift sample code for this use case is coming soon. See the
-[Okta IDX Swift SDK](https://github.com/okta/okta-idx-swift#readme)
-readme for further information.
+The following code can be used in conjuction with the SDK
```

Comment:
Ah, this Sample code section has a little more explanation and context, which is good. As stated earlier, I still think it'd be nice to provide a bit more explanation of what the bits are doing, but this is already much better.

Reply:
I agree. We need to think about how we approach this Swift documentation.

Reply:
Aye. Can be resolved.